src/eth/dependencies.ts

```diff
@@ -1,19 +1,15 @@
-import { JsonRpcProvider } from '@ethersproject/providers';
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
 import { Wallet } from '@ethersproject/wallet';
 import { NonceManager } from '@ethersproject/experimental';
 
-const ethMnemonic = process.env.ETH_MNEMONIC || '';
-const ethRpcUrl = process.env.ETH_RPC_URL || '';
-
 const addressIndicies = {
   // SekhmetDAO
   '0x65e4329e8c0fba31883b98e2cf3e81d3cdcac780': 1
 };
 
-export const provider = new JsonRpcProvider(ethRpcUrl);
-
-export const createWalletProxy = (mnemonic: string) => {
+export const createWalletProxy = (mnemonic: string, chainId: number) => {
   const signers = new Map<string, NonceManager>();
+  const provider = new StaticJsonRpcProvider(`https://rpc.snapshotx.xyz/${chainId}`);
```

Comment:
We can also add the chainId as second parameter, this would reduce number of call.
```suggestion
  const provider = new StaticJsonRpcProvider(`https://rpc.snapshotx.xyz/${chainId}`, chainId);
```